samples/arrays/slice.jakt

```diff
@@ -0,0 +1,25 @@
+/// Expect:
+/// - output: "slice size: 2\n1\n2\n[1, 2]\nslice first: 1\nslice last: 2\nslice contains 2: true\nslice contains 3: false\narray index 0: 0\nslice index 0: 0\n"
+
+function main() {
+    let x = [1, 2, 3, 4, 5]
+    mut slice = x[0..2]
```

Comment:
I like this - out of curiosity what is the type in jakt for these? And can a slice be converted into an array?

I could imagine wanting to do `x[0..2].into_array()` or something if you know where you wanted to slice.

Reply:
This is using `ArraySlice` in the runtime.
Yes, it would be useful to be able to convert it, I can look into that.
Also going to look into extending the range expression to support for example `x[..2]`, `x[2..]`, `x[..]`.
Probably best to get this one finished do those as a new PR.